### PR TITLE
Captcha: form

### PR DESCRIPTION
**Pull request by DishantK1807**
#124 #18

### CODE REVIEW

**Comment by DishantK1807** [comment]
Tested on local and online server

**Comment by DishantK1807** [comment]
captcha_off
Captcha in the registration form

captcha_on
On clicking on the captcha

captcha_error
Error message displayed when captcha is not checked

File: includes/pages/guest_login.php [hunk 1]
```diff
@@ -146,6 +146,28 @@ function guest_register() {
...
+        CURLOPT_URL => 'hppts://www.google.com/recaptcha/api/siteverify',
+        CURLOPT_POST => 1,
+        CURLOPT_POSTFIELDS => [
+          'secret' => '6LeGiyITAAAAAMd--Qw4C3iBPrEM-qZDhQQ4LWMt',
+          'response' => $_REQUEST['g-recaptcha-response'],
+        ]
+      ]);
+      
+      $response = json_decode(curl_exec($curl));
+      $msg .= error(sprintf(_(print_r($response)), $nick), true);
```
Comment (msquare) [inline includes/pages/guest_login.php:165]: You cannot pass a dynamic text (`$response`) through the translation function `_()`.
Does the answer `$response` contain any placeholder to be replaced by the `$nick` name?

File: includes/pages/guest_login.php [hunk 2]
```diff
@@ -146,6 +146,28 @@ function guest_register() {
...
+     * Google reCaptcha Server-Side Handling
+     */
+    if (isset($_REQUEST['g-recaptcha-response']) && !empty($_REQUEST['g-recaptcha-response'])) {
+      $curl = curl_init();
+      curl_setopt_array($curl, [
+        CURLOPT_RETURNTRANSFER => 1,
+        CURLOPT_URL => 'hppts://www.google.com/recaptcha/api/siteverify',
+        CURLOPT_POST => 1,
+        CURLOPT_POSTFIELDS => [
+          'secret' => '6LeGiyITAAAAAMd--Qw4C3iBPrEM-qZDhQQ4LWMt',
```
Comment (msquare) [inline includes/pages/guest_login.php:159]: Move this to `config.php` #142

File: includes/pages/guest_login.php [hunk 3]
```diff
@@ -146,6 +146,28 @@ function guest_register() {
...
       $msg .= error(_("Please enter your Current City."), true);
     }
+    /**
+     * Google reCaptcha Server-Side Handling
+     */
+    if (isset($_REQUEST['g-recaptcha-response']) && !empty($_REQUEST['g-recaptcha-response'])) {
+      $curl = curl_init();
+      curl_setopt_array($curl, [
+        CURLOPT_RETURNTRANSFER => 1,
+        CURLOPT_URL => 'hppts://www.google.com/recaptcha/api/siteverify',
```
Comment (msquare) [inline includes/pages/guest_login.php:156]: Move this to `config.php` #142

File: includes/pages/guest_login.php [hunk 4]
```diff
@@ -146,6 +146,28 @@ function guest_register() {
       $ok = false;
       $msg .= error(_("Please enter your Current City."), true);
     }
+    /**
+     * Google reCaptcha Server-Side Handling
+     */
+    if (isset($_REQUEST['g-recaptcha-response']) && !empty($_REQUEST['g-recaptcha-response'])) {
+      $curl = curl_init();
```
Comment (msquare) [inline includes/pages/guest_login.php:153]: Move reCaptcha calls to a new file `includes/sys_recaptcha.php` containing all needed functions to check with recaptcha. #142

File: includes/sys_template.php [hunk 5]
```diff
@@ -489,4 +489,10 @@ function ReplaceSmilies($neueckig) {
   
   return $neueckig;
 }
+/**
+ * Rendert Google reCaptcha
+ */
+function reCaptcha() {
+  return '<div class="g-recaptcha" data-sitekey="6LeGiyITAAAAAGG2-A9yM47fnB0IRwET_cOunvgf"></div>';
```
Comment (msquare) [inline includes/sys_template.php:496]: Use key from `config.php`! #142 
You may move the function to `sys_recaptcha.php`, but `sys_template.php` is ok too.

File: templates/layout.html [hunk 6]
```diff
@@ -8,6 +8,7 @@
 <link rel="stylesheet" type="text/css" href="vendor/icomoon/style.css" />
 <link rel="stylesheet" type="text/css" href="vendor/bootstrap-datepicker-1.4.0/css/bootstrap-datepicker3.min.css" />
 <script type="text/javascript" src="vendor/jquery-2.1.1.min.js"></script>
+<script src='https://www.google.com/recaptcha/api.js'></script>
```
Comment (msquare) [inline templates/layout.html:11]: Include external libs only if recaptcha is enabled and only on pages where it is used. #142

**Pull request merged**